kip-0038.md

```diff
@@ -0,0 +1,270 @@
+---
+KIP: '0038'
+Title: kadena_getAccounts_v2 API
+Author: Danillo Felixdaal, Bart Huijgen and Albert Groothedde
+Status: Draft
+Type: Standard
+Category: Interface
+Created: 2025-02-17
+---
+
+# kadena_getAccounts_v2 API
+
+## Abstract
+
+This KIP standardizes the `kadena_getAccounts_v2` method, which enables wallets
+to provide a standardized list of accounts they manage. By implementing this
+method, decentralized applications (dApps) can retrieve and present to users the
+available accounts, allowing users to select the desired account(s) for creating
+and signing transactions.
+
+This RPC method allows wallets to return accounts that aren't yet minted
+on-chain, but are managed by the wallet. Therefore the guard is on the level of
+account name, and not on the level of chain account.
+
+## Motivation
+
+The `kadena_getAccounts_v2` method allows retrieval of all accounts managed by a
+wallet along with per fungible and per-chain guard information.
+
+---
+
+## Specification
+
+### Method Definition
+
+**Method Name:** `kadena_getAccounts_v2`
+
+### Request Parameters
+
+No parameters are required.
+
+### Response
+
+The method **MUST** return an array of account objects, even if the wallet only
+supports a single account.
+
+Note that the response can have multiple accounts with the same `accountName`.
+The `guard` is what makes each unique. This allows for dapps to use non-minted
```

Comment:
> The `guard` is what makes each unique.

Is this statement accurate? 

There could be an account `test1` governed by `guardA` and also an account `test2` that is also governed by `guardA`. 

This could happen when working with vanity accounts or accounts associated with contracts that don't enforce principaled (e.g. that account name must correspond to its associated guard) account names.

Reply:
In that case they don't have the same `accountName`, right?

Reply:
The rewording clears it up. Resolving comment.